Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/BuildTarget/Common/TestImpactBuildGraph.h

```diff
@@ -160,6 +211,101 @@ namespace TestImpact
 
         return *vertex;
     }
+
+    template<typename ProductionTarget, typename TestTarget>
+    void BuildGraph<ProductionTarget, TestTarget>::WalkTargetBuildGraphSet(
+        TargetBuildGraphPointer graph,
+        TargetBuildGraphSetPointer set,
+        const BuildTarget<ProductionTarget, TestTarget>& buildTarget,
+        const BuildGraphVertexVisitor<ProductionTarget, TestTarget>& visitor) const
+    {
+        AZStd::queue<const BuildGraphVertex<ProductionTarget, TestTarget>*> vertexQueue;
+        AZStd::unordered_set<const BuildGraphVertex<ProductionTarget, TestTarget>*> visitedVertices;
+        AZStd::unordered_map<const BuildGraphVertex<ProductionTarget, TestTarget>*, size_t> vertexDistances;
+
+        // Skip visiting the root vertex and start visiting its children instead
+        const auto& parentVertex = GetVertexOrThrow(buildTarget);
+        visitedVertices.insert(&parentVertex);
+        for (const auto* child : parentVertex.*graph.*set)
```

Comment:
To make the calling syntax look better.
[AZStd::invoke](https://en.cppreference.com/w/cpp/utility/functional/invoke) can be used instead.
That can return the value of a pointed to member variable

```suggestion
        for (const auto* child : AZStd::invoke(set, AZStd::invoke(graph, parentVertex)))
```

Reply:
Interesting, great suggestion! I will pick this up in the next PR I am putting up 👍